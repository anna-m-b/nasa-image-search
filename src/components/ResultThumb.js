import React from 'react';
import PropTypes from 'prop-types';
import '../styles/ResultThumb.css';

const ResultThumb = ({ thumb, description, id, mediaType }) => {
  const media = mediaType === 'image' ? 'jpg' : 'mp4';
  return (
    <div className="result-thumb">
      <a
        href={`http://images-assets.nasa.gov/${mediaType}/${id}/${id}~orig.${media}`}
      >
        <img
          className="result-thumb_img"
          src={thumb}
          alt={description}
          data-testid="result-image"
          id={id}
        />
      </a>
      <p className="result-thumb_title">{description}</p>
    </div>
  );
};

ResultThumb.propTypes = {
  thumb: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default ResultThumb;


