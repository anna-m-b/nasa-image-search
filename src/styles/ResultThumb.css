.result-thumb {
  margin: 0.5rem;
  width: 100%;
  height: auto;
  display: inline-block;
  position: relative;

}

.result-thumb_img {
  max-width: 100%;
  border-radius: 5px;
  margin-bottom: 0;
}

.result-thumb_title {
  margin: 0;
  background-color: rgba(0, 0, 0, 0.858);
  color: rgba(255, 255, 255, 0.817);
  position: absolute;
  bottom: 1px;
  width: 100%;
  min-height: 0.5rem;
  padding-top: 5px;
  padding-bottom: 5px;
  font-size: 1rem;


}

.result-thumb a {
  margin: 0;
}