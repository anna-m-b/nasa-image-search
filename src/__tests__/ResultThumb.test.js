import ResultThumb from '../components/ResultThumb';
import { render } from '@testing-library/react';

describe('ResultThumb', () => {
  const validProps = {
    thumb: 'https://via.placeholder.com/200',
    description: 'test title',
    id: 'P12345',
    mediaType: 'image'
  };
  test('it renders correctly', () => {
    const { asFragment } = render(<ResultThumb {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('it renders props correctly', () => {
    beforeEach(() => {
      render(<ResultThumb {...validProps} />);
    });
    test('image src is set to thumb prop', () => {
      const image = document.querySelector('img');
      expect(image.src).toEqual(validProps.thumb);
    });
    test('image alt text is set to description prop', () => {
      const image = document.querySelector('img');
      expect(image.alt).toEqual(validProps.description);
    });
    test('anchor tag href contains id prop', () => {
      const anchor = document.querySelector('a');
      expect(anchor.href).toEqual(
        `http://images-assets.nasa.gov/image/${validProps.id}/${validProps.id}~orig.jpg`
      );
    });
  });
});
