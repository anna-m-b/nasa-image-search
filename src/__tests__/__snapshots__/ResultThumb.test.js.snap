// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultThumb it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="result-thumb"
  >
    <a
      href="http://images-assets.nasa.gov/image/P12345/P12345~orig.jpg"
    >
      <img
        alt="test title"
        class="result-thumb_img"
        data-testid="result-image"
        id="P12345"
        src="https://via.placeholder.com/200"
      />
    </a>
    <p
      class="result-thumb_title"
    >
      test title
    </p>
  </div>
</DocumentFragment>
`;
